scipy/stats/_continuous_distns.py

```diff
@@ -3261,6 +3262,313 @@ def _ppf(self, q, k, s):
+    | h    | k=0.0       | k=1.0          | -inf<=k<=inf  |
+    +======+=============+================+===============+
+    | -1.0 | Logistic    |                | Generalized   |
+    |      |             |                | Logistic(1)   |
+    +------+-------------+----------------+---------------+
+    |  0.0 | Gumbel      | Reverse        | Generalized   |
+    |      |             | Exponential(2) | Extreme Value |
+    +------+-------------+----------------+---------------+
+    |  1.0 | Exponential | Uniform        | Generalized   |
+    |      |             |                | Pareto        |
+    +------+-------------+----------------+---------------+
+
+    (1) This test doesn't pass.
+    (2) Not part of kappa4 tests since this distribution is not in
+        scipy.
+
+    References
+    ----------
+    J.C. Finney, "Optimization of a Skewed Logistic Distribution With Respect
+    to the Kolmogorov-Smirnov Test", A Dissertation Submitted to the Graduate
+    Faculty of the Louisiana State University and Agricultural and Mechanical
+    College, (August, 2004),
+    http://etd.lsu.edu/docs/available/etd-05182004-144851/unrestricted/
+    Finney_dis.pdf
+
+    B. Kumphon, A. Kaew-Man, P. Seenoi, "A Rainfall Distribution for the Lampao
+    Site in the Chi River Basin, Thailand", Journal of Water Resource and
+    Protection, vol. 4, 866-869, (2012).
+    http://file.scirp.org/pdf/JWARP20121000009_14676002.pdf
+
+    C. Winchester, "On Estimation of the Four-Parameter Kappa Distribution", A
+    Thesis Submitted to Dalhousie University, Halifax, Nova Scotia, (March
+    2000).
+    http://www.nlc-bnc.ca/obj/s4/f2/dsk2/ftp01/MQ57336.pdf
+
+    %(after_notes)s
+
+    %(example)s
+
+    """
+    def _argcheck(self, h, k):
+        condlist = [np.logical_and(h > 0, k > 0),
+                    np.logical_and(h > 0, k == 0),
+                    np.logical_and(h > 0, k < 0),
+                    np.logical_and(h <= 0, k > 0),
+                    np.logical_and(h <= 0, k == 0),
+                    np.logical_and(h <= 0, k < 0)]
+
+        def f0(h, k):
+            return (1.0 - h**(-k))/k
+
+        def f1(h, k):
+            return log(h)
+
+        def f3(h, k):
+            a = np.empty(h.shape)
+            a[:] = -inf
+            return a
+
+        def f5(h, k):
+            return 1.0/k
+
+        self._a = _lazyselect(condlist,
+                             [f0, f1, f0, f3, f3, f5],
+                             [h, k],
+                             default=nan)
+
+        def f0(h, k):
+            return 1.0/k
+
+        def f1(h, k):
+            a = np.empty(h.shape)
+            a[:] = inf
+            return a
+
+        self._b = _lazyselect(condlist,
+                             [f0, f1, f1, f0, f1, f1],
+                             [h, k],
+                             default=nan)
+        return (h == h)
+
+    def _pdf(self, x, h, k):
+        cond = np.logical_and(x >= self._a, x <= self._b)
```

Comment:
Once `self._a` is renamed to `self.a`, this condition is no longer needed I believe, because it's handled in the generic `pdf`.